### PR TITLE
Merge detached entities before persist/delete

### PR DESCRIPTION
**Pull request by javier-godoy**
(and two other minor fixes)

### CODE REVIEW

**Comment by javier-godoy** [comment]
Maybe we need to replace `saveOrUpdate` (which is an hibernatism) with `update` (at least in the Data layer). 

https://github.com/AppJars/commons-backend/blob/a8d4d42cc34497022196d5c4b23c9387d5f9e560/commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java#L89-L93

File: commons-model/src/main/java/com/appjars/saturn/model/BaseEntity.java [hunk 1]
```diff
@@ -64,5 +65,14 @@ public boolean equals(Object obj) {
 		}
 		return true;
 	}
+	
+	@Override
+	public String toString() {
+		if (id!=null) {
+			return this.getClass().getName()+"@"+id;
+		} else {
+			return this.getClass().getName()+"@transient;"+Integer.toHexString(System.identityHashCode(this));
```
Comment (mlopezFC) [inline commons-model/src/main/java/com/appjars/saturn/model/BaseEntity.java:74]: Maybe something silly, but why semicolon? maybe with colon instead it's nicer:
`com.test.model.MyPojo@transient:2101973421`
vs
`com.test.model.MyPojo@transient;2101973421`

File: commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java [hunk 2]
```diff
@@ -72,13 +76,31 @@ default K save(S entity) {
 	@Override
 	default void saveOrUpdate(S entity) {
```
Comment (mlopezFC) [inline commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java:77]: With this implementation, my understanding is that basically what this method does is to "update" the entity (it won't "save" a new entity). So maybe we should consider change its name from `saveOrUpdate` to just `update`.

File: commons-model/src/main/java/com/appjars/saturn/model/BaseEntity.java (hunk 1 above)
Comment (javier-godoy) [reply commons-model/src/main/java/com/appjars/saturn/model/BaseEntity.java]: Done ✔

File: commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java (hunk 2 above)
Comment (javier-godoy) [reply commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java]: Done ✔

**Review by mlopezFC — APPROVED** [review]
Excellent, now it feels more consistent. And I also like the new handling of ids in BaseEntity,

**Pull request merged**